spec/latest/core/core.jsonld

```diff
@@ -485,7 +491,7 @@
       "@type": "rdf:Property",
       "label": "mapping",
       "comment": "A variable-to-property mapping of the IRI template.",
-      "domain": "hydra:IriTemplate",
+      "domain": "hydra:Template",

```

Comment:
Regardless of the other template changes, I think this should remain an IRI Template. 

Or to test the logic, does it make sense to have a `HeaderTemplate` used with `rdf:Property`?

Reply:
Well, the `hydra:HeaderTemplate` would also need a `hydra:mapping`. I've just noticed that the range should be also decoupled from IriTemplateMapping.

The thing is, IriTemplate implies ... an IRI, which is not the case here. For headers (or body in future??) that would be completely different templating mechanism (i.e. razor :P)